### PR TITLE
add mish trt plugin

### PR DESCRIPTION
**Pull request by wangxinxin08**
### PR types
New features

### PR changes
Others

### Describe
add mish trt plugin, the inference speed of ppyolov2 r50vd dcn before and after this PR can be seen as follows:
f349a7764059f721ad5fd4e48aec0aba

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline paddle/fluid/inference/tensorrt/convert/mish_op.cc]: 2020 -> 2021

Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/convert/mish_op.cc]: done, thx!

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+#endif
+}
+
+template <typename T>
+__device__ T kSoftplus(T x, T threshold) {
+  if (x > threshold) {
+    return x;
+  } else {
+    return log(exp(x) + static_cast<T>(1.0f));
+  }
```
Comment (qingqing01) [inline paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: better 
```
 x = x > threshold ? x : log(...)
```

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+template <>
+__device__ half kSoftplus<half>(half x, half threshold) {
+#if CUDA_ARCH_FP16_SUPPORTED(__CUDA_ARCH__)
+  if (x > threshold) {
+    return x;
+  } else {
+    return hlog(hexp(x) + static_cast<half>(1.0f));
+  }
```
Comment (qingqing01) [inline paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: same as above

**Review by qingqing01 — COMMENTED** [review]
@Superjomn @shangzhizhou  Please also review. 

Is there unit test for trt plugin?

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+class MishOpConverter : public OpConverter {
+ public:
+  void operator()(const framework::proto::OpDesc& op,
+                  const framework::Scope& scope, bool test_mode) override {
+    VLOG(4) << "convert fluid Mish op to tensorrt Mish plugin";
+
+    framework::OpDesc op_desc(op, nullptr);
+    // Declare inputs
+    int input_num = op_desc.Input("X").size();
+    PADDLE_ENFORCE_EQ(input_num, 1,
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/convert/mish_op.cc]: 关于输入数量的判断请前置到op_teller中，降低推理报错的可能性。

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                      platform::errors::InvalidArgument(
+                          "Mish op has only 1 input, but got %d", input_num));
+    auto* input = engine_->GetITensor(op_desc.Input("X")[0]);
+    // Get output
+    size_t output_num = op_desc.Output("Out").size();
+    PADDLE_ENFORCE_EQ(output_num, 1,
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/convert/mish_op.cc]: 同上

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                          "Mish op has only 1 output, but got %d", output_num));
+
+    const float threshold =
+        op_desc.HasAttr("threshold")
+            ? BOOST_GET_CONST(float, op_desc.GetAttr("threshold"))
+            : 20.0f;
+
+    nvinfer1::ILayer* layer = nullptr;
+    if (engine_->with_dynamic_shape()) {
+#if IS_TRT_VERSION_GE(6000)
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/convert/mish_op.cc]: 目前已经不在支持trt6之前的版本，可以假设trt最低版本6

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+namespace inference {
+namespace tensorrt {
+namespace plugin {
+
+int MishPlugin::initialize() { return 0; }
+
+nvinfer1::Dims MishPlugin::getOutputDimensions(int index,
+                                               const nvinfer1::Dims* in_dims,
+                                               int nb_inputs) {
+  assert(nb_inputs == 1);
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: 请使用PADDLE_ENFORCE的形式

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.h [hunk 8]
```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.h:1]: 请添加单测，覆盖不同精度、输入属性、序列化等功能，可参考https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/unittests/ir/inference/test_trt_pool_op.py

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu (hunk 2 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: done, thx!

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu (hunk 3 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: done, thx!

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc (hunk 4 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/convert/mish_op.cc]: done, thx!

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc (hunk 5 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/convert/mish_op.cc]: done, thx!

File: paddle/fluid/inference/tensorrt/convert/mish_op.cc (hunk 6 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/convert/mish_op.cc]: done, thx!

File: paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu (hunk 7 above)
Comment (wangxinxin08) [reply paddle/fluid/inference/tensorrt/plugin/mish_op_plugin.cu]: done, thx!

**Review by shangzhizhou — DISMISSED** [review]
LGTM，请本地验证下TensorRT8的编译

**Review by jiweibo — APPROVED** [review]
LGTM for mish

**Pull request merged**